src/Strategy/Swift/PackageResolved.hs

```diff
@@ -0,0 +1,80 @@
+module Strategy.Swift.PackageResolved (
+  SwiftPackageResolvedFile (..),
+  SwiftResolvedPackage (..),
+  resolvedDependenciesOf,
+) where
+
+import Data.Aeson (
+  FromJSON (parseJSON),
+  Object,
+  withObject,
+  (.:),
+  (.:?),
+ )
+import Data.Aeson.Types (Parser)
+import Data.Foldable (asum)
+import Data.Text (Text)
+import DepTypes (DepType (GitType), Dependency (..), VerConstraint (CEq))
+
+data SwiftPackageResolvedFile = SwiftPackageResolvedFile
+  { version :: Integer
+  , pinnedPackages :: [SwiftResolvedPackage]
+  }
+  deriving (Show, Eq, Ord)
+
+data SwiftResolvedPackage = SwiftResolvedPackage
+  { package :: Text
+  , repositoryURL :: Text
+  , repositoryBranch :: Maybe Text
+  , repositoryRevision :: Maybe Text
+  , repositoryVersion :: Maybe Text
+  }
+  deriving (Show, Eq, Ord)
+
+instance FromJSON SwiftPackageResolvedFile where
+  parseJSON = withObject "Package.resolved content" $ \obj -> do
+    version <- obj .: "version"
+    pinnedPackages <- obj .: "object" |> "pins"
+    pure $ SwiftPackageResolvedFile version pinnedPackages
```

Comment:
[nit]: Is there a reason we need do notation for this?  If there is, I'm not seeing it.  I think we could rewrite this to be applicative.

```suggestion
instance FromJSON SwiftPackageResolvedFile where
  parseJSON = withObject "Package.resolved content" $ \obj ->
    SwiftPackageResolvedFile <$> obj .: "version"
      <*> (obj .: "object" |> "pins")
```